### DISCUSSION TITLE
sequel + snowflake odbc - how to set connection pool size?

### DISCUSSION THREAD

**Discussion by LiChoi** [jeremyevans/sequel#1981] [General]
Using sequel gem to connect to snowflake using snowflake's odbc driver.

DB = Sequel.odbc CONNECTION_PARAMETERS

Where CONNECTION_PARAMETERS contains the following parameters required by the odbc driver: driver, server, warehouse, database, schema, uid, pwd

DB.pool.size returns 1 - and I believe that the number of connections available in the pool is truly and only one because this has resulted in Sequel::PoolTimeout errors under heavy load. This issue goes away if I create a new DB connection object each time I need to make a query (and I make sure to close the connection after each use).

The workaround works but I'd rather have a proper connection pool set up. Setting "max_connections" in the parameters to a number greater than 1 doesn't change anything. Not sure how to change the pool size using sequel with the snowflake odbc connection parameters.

**Is the connection pool parameter handled by sequel gem? Or this specific to the snowflake odbc driver? Any ideas how to set the connection pool size?**

**Comment by jeremyevans** [comment]
The connection pool is handled by Sequel, not specific to any adapter/driver. The connection pool size is the current number of connections in the pool.  It isn't the maximum size.  Connections are created as needed until the maximum size is reached.  The default connection pool maximum size is 4. Since increasing the :max_connections option doesn't help, there is probably something else wrong, and you should post a minimal self contained reproducible example showing the problem.

**Reply by LiChoi** [reply]
Thank you for responding and clarifying what DB.pool.size means.

My mistake. I've been using pool.size to verify whether or not the "max_connections" parameter was applied. Since DB.pool.size is the current number of connections (not the max), that explains why it is always 1 in my local environment.

I realize now that I should have been checking DB.pool.max_size, which returns 4, the default.

I will set the max_connections to a bigger number and do a load test to see if pool timeout issue has resolved.

Thanks again.
